packages/loader/src/gltf/parser/GLTFAnimationParser.ts

```diff
@@ -210,56 +210,33 @@ export class GLTFAnimationParser extends GLTFParser {
     }
   }
 
-  parse(context: GLTFParserContext): AssetPromise<AnimationClip[]> | void {
-    const { glTF, glTFResource } = context;
-    const { entities } = glTFResource;
-    const { animations, accessors, bufferViews } = glTF;
-    if (!animations) {
-      return;
-    }
-    const animationClipsPromiseInfo = context.animationClipsPromiseInfo;
+  parse(context: GLTFParserContext, index: number): Promise<AnimationClip> {
+    const animations = context.glTF.animations;
+    if (!animations) return Promise.resolve(null);
 
-    const animationClipCount = animations.length;
-    const animationClipPromises = [];
-    const animationsIndices = new Array<{
-      name: string;
-      index: number;
-    }>(animationClipCount);
+    return this._parseSingleAnimation(context, animations[index], index);
+  }
 
-    let parseStandardPropertyPromises = new Array<Promise<void>>();
-    for (let i = 0; i < animationClipCount; i++) {
-      const animationInfo = animations[i];
-      const { name = `AnimationClip${i}` } = animationInfo;
+  private _parseSingleAnimation(
+    context: GLTFParserContext,
+    animationInfo: IAnimation,
+    index: number
+  ): Promise<AnimationClip> {
+    const { name = `AnimationClip${index}` } = animationInfo;
 
-      let animationClip = <AnimationClip>(
-        GLTFParser.executeExtensionsCreateAndParse(animationInfo.extensions, context, animationInfo)
-      );
+    let animationClip = <Promise<AnimationClip> | AnimationClip>(
+      GLTFParser.executeExtensionsCreateAndParse(animationInfo.extensions, context, animationInfo)
+    );
 
-      if (!animationClip) {
-        animationClip = new AnimationClip(name);
-        parseStandardPropertyPromises.push(
-          GLTFAnimationParser._parseStandardProperty(context, animationClip, animationInfo)
-        );
-      }
+    let parseStandardPropertyPromise;
 
-      animationClipPromises.push(animationClip);
+    if (!animationClip) {
+      animationClip = new AnimationClip(name);
+      parseStandardPropertyPromise = GLTFAnimationParser._parseStandardProperty(context, animationClip, animationInfo);
     }
-
-    return AssetPromise.all(parseStandardPropertyPromises).then(() => {
-      return AssetPromise.all(animationClipPromises).then((animationClips) => {
-        glTFResource.animations = animationClips;
-        for (let i = 0; i < glTF.animations.length; i++) {
-          const animationInfo = glTF.animations[i];
-          GLTFParser.executeExtensionsAdditiveAndParse(
-            animationInfo.extensions,
-            context,
-            animationClips[i],
-            animationInfo
-          );
-        }
-        animationClipsPromiseInfo.resolve(animationClips);
-        return animationClipsPromiseInfo.promise;
-      });
+    return Promise.all([animationClip, parseStandardPropertyPromise]).then(([animationClip]) => {
```

Comment:
Very poorly readable logic， just want less code? Not good

Reply:
done